crates/frontend/src/components/monaco_editor.rs

```diff
@@ -0,0 +1,58 @@
+use std::{cell::RefCell, rc::Rc};
+
+use leptos::*;
+use monaco::api::CodeEditor;
+use monaco::sys::editor::{IEditorMinimapOptions, IStandaloneEditorConstructionOptions};
+#[derive(Debug, Clone, strum_macros::Display)]
+#[strum(serialize_all = "lowercase")]
+pub enum Languages {
+    Javascript,
+    Json,
+}
+
+pub type EditorModelCell = Rc<RefCell<Option<CodeEditor>>>;
```

Comment:
@Datron any specific reason to use Rc<RefCell>. , we can use just Option<CodeEditor> also